src/unix/thread.c

```diff
@@ -875,3 +890,50 @@ void uv_key_set(uv_key_t* key, void* value) {
   if (pthread_setspecific(*key, value))
     abort();
 }
+
+
+int uv__thread_setname(const char* name) {
+  char namebuf[UV_PTHREAD_MAX_NAMELEN_NP];
+  strncpy(namebuf, name, sizeof(namebuf) - 1);
+  namebuf[sizeof(namebuf) - 1] = '\0';
+#if defined(__APPLE__)
+  int err = pthread_setname_np(namebuf);
+  if (err)
+    return UV__ERR(errno);
+  return 0;
+#elif defined(__NetBSD__)
+  return UV__ERR(pthread_setname_np(pthread_self(), "%s", namebuf));
+#elif defined(__OpenBSD__)
+  pthread_set_name_np(pthread_self(), namebuf);
+  return 0;
+#else
+  return UV__ERR(pthread_setname_np(pthread_self(), namebuf));
+#endif
+}
+
+
+int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
+#if defined(__ANDROID_API__) && __ANDROID_API__ < 26
+  return UV_ENOSYS;
+#else
+  int r;
+  char thread_name[UV_PTHREAD_MAX_NAMELEN_NP];
+#if defined(__OpenBSD__)
+  pthread_get_name_np(*tid, thread_name, sizeof(thread_name));
+  r = 0;
+#else
+  r = pthread_getname_np(*tid, thread_name, sizeof(thread_name));
+#endif
+  if (r != 0) {
+#if defined(__APPLE__)
+    return UV__ERR(errno);
+#else
+    return UV__ERR(r);
+#endif
+#endif /* defined(__ANDROID_API__) && __ANDROID_API__ < 26 */
+  }
+
+  strncpy(name, thread_name, size - 1);
+  name[size - 1] = '\0';
+  return 0;
```

Comment:
Syntax error on Android:
```suggestion
  }

  strncpy(name, thread_name, size - 1);
  name[size - 1] = '\0';
  return 0;
#endif /* defined(__ANDROID_API__) && __ANDROID_API__ < 26 */
```
Honestly, this ifdef soup is a little over the top. The fact you, its author, got it wrong suggests other people will find it even more challenging. Consider rewriting it like this:
```c
#if defined(__ANDROID_API__) && __ANDROID_API__ < 26
int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
  return UV_ENOSYS;
}
#elif defined(__APPLE__)
int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
  // ...
}
#elif defined(__OpenBSD__)
int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
  // ...
}
#else
int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
  // ...
}
#endif
```
More duplication but much more legible.

Reply:
Sounds good